sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestDefineInParameter.java

```diff
@@ -0,0 +1,154 @@
+
+        assertThat(testDao.findById(testColumns, "test", 1)).isEqualTo(something);
+        assertThat(testDao.findById(testColumns, "testNullable", 1)).isNull();
+
+        assertThat(testDao.findById(testColumns, "test", 2)).isNull();
+        assertThat(testDao.findById(testColumns, "testNullable", 2)).isEqualTo(nothing);
+    }
+
+    @Test
+    public void testWithBindBean() throws Exception
+    {
+        TestDao testDao = handle.attach(TestDao.class);
+
+        Something something = new Something(1, "Some Pig");
+        Something nothing = new Something(2, null);
+
+        testDao.insert("test", testColumns, something);
+        testDao.insert("testNullable", testColumns, nothing);
+
+        assertThat(testDao.findById(testColumns, "test", 1)).isEqualTo(something);
+        assertThat(testDao.findById(testColumns, "testNullable", 1)).isNull();
+
+        assertThat(testDao.findById(testColumns, "test", 2)).isNull();
+        assertThat(testDao.findById(testColumns, "testNullable", 2)).isEqualTo(nothing);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testIllegalArray()
+    {
+        TestDao testDao = handle.attach(TestDao.class);
+        String[] columnsArray = {"id", "name"};
+        List<Object> values = new ArrayList<>();
+        values.add(1);
+        values.add("Some Pig");
+        testDao.badInsert("test", columnsArray, values);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testUnsupportedInsert()
+    {
+        TestDao testDao = handle.attach(TestDao.class);
+        List<Object> values = new ArrayList<>();
+        values.add(1);
+        values.add("Some Pig");
+        testDao.unsupportedInsert("test", testColumns, values);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEmptyList()
+    {
+        TestDao testDao = handle.attach(TestDao.class);
+        List<String> noColumns = new ArrayList<>();
+        Something something = new Something(1, "Some Pig");
+        testDao.insert("test", noColumns, something);
+    }
+
+    @RegisterRowMapper(SomethingMapper.class)
+    public interface TestDao
+    {
+        @SqlUpdate("insert into <table> (<columns>) values (<values>)")
+        void insert(@Define("table") String table, @DefineIn("columns") List<String> columns, @BindIn("values") List<Object> values);
+
+        @SqlUpdate("insert into <table> (<columns>) values (:id, :name)")
+        void insert(@Define("table") String table, @DefineIn("columns") List<String> columns, @BindBean Something s);
+
+        @SqlUpdate("insert into <table> (<columns>) values (<values>)")
+        void badInsert(@Define("table") String table, @DefineIn("columns") Object columns, @BindIn("values") List<Object> values);
+
+        @SqlUpdate("insert into <table> (<columns>) values (<values>)")
+        void unsupportedInsert(@Define("table") String table, @DefineIn List<String> columns, @BindIn("values") List<Object> values);
```

Comment:
I guess I had thought that was the expected behavior of @Define, but I was misunderstanding the way Parameter works. Not used to working in Java 8! 

Now that I look at it again, it should only be happening when it's passed a raw value rather than a named variable. I have to wonder why this test was passing on my machine.

This will still be useful with some modification though, since it will be accepting arrays on the next iteration.